crates/circuit/src/vf2.rs

```diff
@@ -955,17 +963,14 @@ struct State<G: GraphBase, OtherId> {
     /// not yet mapped, the other index is stored as `I::max()`.
     mapping: Vec<OtherId>,
     /// Mapping from node index to the generation at which a node was first added to the mapping
-    /// that had an outbound edge to that index.  This can be used to find new candidate nodes to
-    /// add to the mapping; you typically want your next node to be one that has edges linking it to
-    /// the existing mapping (but isn't yet _in_ the mapping).
-    out: Vec<usize>,
-    /// Same as `out`, except we're tracking the nodes that have an edge from them _into_ the
-    /// mapping.  This isn't used if the graph is undirected, since it'd just duplicate `out`.
-    ins: Vec<usize>,
-    /// The number of non-zero entries in `out`.
-    out_size: usize,
-    /// The number of non-zero entries in `in`.  This is always zero for undirected graphs.
-    ins_size: usize,
+    /// that had an `[outbound, inbound]` edge to that index.  This can be used to find new
+    /// candidate nodes to add to the mapping; you typically want your next node to be one that has
+    /// edges linking it to the existing mapping (but isn't yet _in_ the mapping).
+    neighbor_since: Vec<[Option<NonZero<usize>>; 2]>,
```

Comment:
Both: a much better name than before; and I have learned about the cool `NonZero` struct.

Reply:
The advantage of `NonZero` is specifically that it lets `Option<NonZero>>` be stored by the zero value of the type, so the option takes up no additional space.